pinot-core/src/main/java/com/linkedin/pinot/core/query/pruner/PartitionSegmentPruner.java

```diff
@@ -32,6 +34,7 @@
  * Implementation of {@link SegmentPruner} that uses partition information to perform pruning.
  */
 public class PartitionSegmentPruner extends AbstractSegmentPruner {
+  public static final String IN_CLAUSE_DELIMITER = "\t\t";
```

Comment:
I don't see how this change can improve latency from 5seconds down to 80ms. Without this change, the only cost to pay is for query planning, where planner should have bailed out determining that the predicate cannot be satisfied at all. Pruner only helps with throughput by avoiding the scheduling of threads for planning. Also, if the the planner did not bail out early, still 5 seconds seems super high, in which case we should understand where the time is being spent and fix that as well.